maven_iuvs/echelle.py

```diff
@@ -841,3 +880,1530 @@ def find_files_with_geometry(file_index):
+
+    sqrt( sum((M-Fmed)^2) / ((2*Wdt+1)^2))
+
+    Where M is an array defined by a sliding (2*Wdt+1)x(2*Wdt+1) window, Fmed is the median value within that window, 
+    and Wdt is the radius of the window not including the center pixel. Believe it or not, it's not easy to do that with built-in 
+    functions, which all seem to reduce the size of your data when using the sliding window. This function lets us do the 
+    calculation at every pixel without reducing the size of the data in a vectorized way, avoiding three nested for loops.
+
+    Parameters
+    ----------
+    data : array (2D)
+           Detector image for one integration
+    Wdt : int
+         1/2 the window width, i.e. the "radius". Window side length will be 2*Wdt + 1.
+    padmode : string
+              Values must be generated for regions outside the edge. This mode gets passed to np.pad.
+
+    Returns
+    ----------
+    Fsig : array
+           The calculation sqrt( sum((M-Fmed)^2) / ((2*Wdt+1)^2)) at every pixel. 
+    """
+    # ONLY DEFINED FOR 2D
+    # Find the proper size of a:
+    n = data.shape[0]  # rows
+    m = data.shape[1]  # columns
+
+    window_edge = 2*Wdt + 1
+    # print(f"Sliding window dimensions are [{window_edge_size}, {window_edge_size}]")
+
+    # Pad edges with 0's, to handle edge cases.
+    data_pad = np.pad(data, pad_width=Wdt, mode=padmode, constant_values=np.nan)
+
+    # Get the sliding window shape
+    # Note that the array 'a' is trimmed to be (n-2, m-2) because sliding_window_view can't handle edges.
+    # Since we padded it we are golden
+    window = sliding_window_view(data_pad, (window_edge, window_edge)) 
+    # print(f"Sliding window dimensions are {window.shape}")
+
+    # Operating on data, calculate the median values in a window_edge x window_edge box and make sure they are the same shape as the trimmed array.
+    # axes 2,3 is hard coded because this only works for 2D
+    window_medians = np.nanmedian(window, axis=(2,3)).reshape((window.shape[0], window.shape[1], 1, 1)) 
+
+    # Subtract the medians from the views
+    subtracted_med = np.subtract(window, window_medians)
+    # print(f"Dimensions of subtracted_med are {subtracted_med.shape}")
+    
+    # get sum of squares
+    sumsq = np.nansum(subtracted_med**2, axis=(2,3))    
+    # print(f"Dimensions of sumsq are {sumsq.shape}")
+
+    # Get Fsig
+    Fsig = np.sqrt(sumsq / (window_edge**2))
+    # print(f"Dimensions of array are {a.shape[0]}x{a.shape[1]}")
+    # print(f"Dimensions of Fsig are {Fsig.shape[0]}x{Fsig.shape[1]}")
+
+    return Fsig
+
+
+# Fitting plots because they don't work in echelle_Graphics because of circular import
+
+def plot_line_fit(data_wavelengths, data_vals, model_fit, fit_params_for_printing, data_unc=None, H_a=0, H_b=0, D_a=0, D_b=0, t="Fit", unit="DN", show_integration_regions=False, logview=False, plot_bg=True):
```

Comment:
should plotting functions be in echelle_graphics?

Reply:
Yes, definitely, but I left them here to avoid a circular import problem. Upon review, I think I can fix the problem so that they can be moved to the right place. Stand by.

Reply:
Fixed by [dd078e8](https://github.com/lasp/maven_iuvs/pull/66/commits/dd078e895dc4a009322e3eb6dbe764070be49269)